File: Algorithms/Python/Climbing-Stairs.py
class Solution(object):

    def climbStairs(self, n):
        """
        :type n: int
        :rtype: int
        """
        ans = [1, 2]
        for i in range(2, n):
            ans.append(ans[i - 2] + ans[i - 1])
        return ans[n - 1]

File: Algorithms/Java/Climbing Stairs.java
public class Solution {
    public int climbStairs(int n) {
    	if (n < 2)
			return 1;
		int[] s = new int[n];
		s[0] = 1;
		s[1] = 2;
		for (int i = 2; i < n; i++)
			s[i] = s[i - 1] + s[i - 2];
		return s[n - 1]; 
    }
}